### PR TITLE
Classes

### PR DESCRIPTION
**Pull request by Dadadon**
**Created class Person with the following**
 - Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? that returns true if person is of age or if they have permission from parents.

**Created class Student with the following**
- Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".

**Created class Teacher with the following**
- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.

### CODE REVIEW

**Comment by sja-thedude** [comment]
Hi @Dadadon,

You have linter errors, you can fix with `rubocop -A `. I will wait for another 5 mins. If you fix I will proceed else invalidate it.

**Comment by sja-thedude** [comment]
Hey @Dadadon, For the Gemfile error, kindly make a new file called Gemfile and add the below then run on your terminal bundle install.
```
source 'https://rubygems.org'

git_source(:github) { |repo_name| "https://github.com/#{repo_name}" }
gem 'rubocop', '>= 1.0', '< 2.0'

# gem "rails"
```

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Invalid Code Review Request ❌

Hi @Dadadon, This is @sja-thedude your code reviewer.

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

You have submitted a project, that has linter error, I tried to drop a comment on GitHub about how to fix it and also tried to contact you on slack but you haven't responded so invalidating.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: Classes/person.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def generate_id
+    # code to generate a unique ID
+  end
```
Comment (codepantha) [inline Classes/person.rb]: - [ ] Please implement this method to generate a random unique id because you have called it in the `initialize` method 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Dadadon 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Created classes 📌 
✔️ No linter errors
✔️ Used GitFlow correctly

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Dadadon 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**